src/tests/system/tests/test_netgroups.py

```diff
@@ -309,3 +312,152 @@ def test_netgroup__uid_gt_2147483647(client: Client, provider: GenericProvider):
+    """
+    :title: SSSD does not crash in nss responder after netgroup timeout when backend is offline
+    :setup:
+        1. A user (user-1) and a netgroup (ng-1) are created, and the user is added as a member of the netgroup
+        2. SSSD is restarted with a clean configuration to ensure no stale data is present
+    :steps:
+        1. Depending on the type of provider (AD, IPA, Samba, or LDAP), the script updates the SSSD configuration
+            with an incorrect server URI (e.g., typo.dc.hostname)
+        2. SSSD is restarted again to apply the new configuration
+        3. Checks the status of the SSSD domain to ensure it is offline due to the misconfigured server
+        4. Capture the process ID (PID) of the sssd_nss process
+        5. Try to retrieve the netgroup information again, expecting it to fail since the SSSD domain is offline
+        6. Verify that the SSSD processes (sssd, sssd_be, sssd_nss, sssd_pam) are still running and that the
+            sssd_nss process ID has not changed, indicating that SSSD has not crashed or restarted unexpectedly
+    :expectedresults:
+        1. SSSD configured with incorrect server backend
+        2. SSSD restarted
+        3. SSSD Offline
+        4. Pid of sssd_nss captured
+        5. Netgroup info cant be retrieve
+        6. SSSD not crashed
+    :customerscenario: True
+    """
+    user = provider.user("user-1").add()
+    netgroup = provider.netgroup("ng-1").add().add_member(user=user)
+    client.sssd.restart(clean=True)
+
+    result = client.tools.getent.netgroup(netgroup.name)
+    assert result is not None
+    assert result.members[0].user == "user-1"
+
+    hostname = client.host.hostname
+    if isinstance(provider, (AD)):
+        bad_ldap_uri = "typo.dc.%s" % hostname
+        client.sssd.dom("test").update(ad_server=bad_ldap_uri)
+
+    if isinstance(provider, (IPA)):
+        bad_ldap_uri = "typo.master.%s" % hostname
+        client.sssd.dom("test").update(ipa_server=bad_ldap_uri)
+
+    if isinstance(provider, (Samba)):
+        bad_ldap_uri = "typo.dc.%s" % hostname
+        client.sssd.dom("test").update(ad_server=bad_ldap_uri)
+
+    if isinstance(provider, (LDAP)):
+        bad_ldap_uri = "ldaps://typo.%s" % hostname
+        client.sssd.dom("test").update(ldap_uri=bad_ldap_uri)
+
+    client.sssd.restart(clean=True)
+
+    # Check backend status
+    assert "Offline" in client.host.conn.run("sssctl domain-status test -o").stdout
+
+    pid_nss = "pidof sssd_nss"
+    pid_nss1 = client.host.conn.run(pid_nss).stdout
+
+    # request for netgroup
+    assert not client.tools.getent.netgroup(netgroup.name)
+
+    sssd_proc = ["sssd", "sssd_be", "sssd_nss", "sssd_pam"]
+    for proc in sssd_proc:
+        pgrep = "pgrep %s" % proc
+        client.host.conn.run(pgrep)
+
+    pid_nss2 = client.host.conn.run(pid_nss).stdout
+    assert pid_nss1 == pid_nss2
+
+
+@pytest.mark.importance("low")
+@pytest.mark.ticket(bz=1406437)
```

Comment:
I think that we can drop the allocation error check and if everything else is covered elsewhere drop it altogether.